com.unity.netcode.adapter.utp/Tests/Runtime/TransportTests.cs

```diff
@@ -128,19 +127,24 @@ public IEnumerator PingPongSimultaneous([ValueSource("k_DeliveryParameters")] Ne
         [UnityTest]
         public IEnumerator SendMaximumPayloadSize([ValueSource("k_DeliveryParameters")] NetworkDelivery delivery)
         {
-            InitializeTransport(out m_Server, out m_ServerEvents);
-            InitializeTransport(out m_Client1, out m_Client1Events);
+            // We want something that's over the old limit of ~44KB for reliable payloads.
+            var payloadSize = 64 * 1024;
+
+            InitializeTransport(out m_Server, out m_ServerEvents, payloadSize);
+            InitializeTransport(out m_Client1, out m_Client1Events, payloadSize);
 
             m_Server.StartServer();
             m_Client1.StartClient();
 
             yield return WaitForNetworkEvent(NetworkEvent.Connect, m_Client1Events);
 
-            var payload = new ArraySegment<byte>(new byte[UnityTransport.InitialMaxPayloadSize]);
+            var payload = new ArraySegment<byte>(new byte[payloadSize]);
             m_Client1.Send(m_Client1.ServerClientId, payload, delivery);
 
             yield return WaitForNetworkEvent(NetworkEvent.Data, m_ServerEvents);
 
+            Assert.AreEqual(payloadSize, m_ServerEvents[1].Data.Count);
```

Comment:
This checks that a payload of the correct size is delivered. However, what if there was a bug where "1111....12222....23333....3" was delivered as "1111....11111....11111....1" (repeated fragment blocks) ?
The test would miss it. I'm not saying there's an issue, but I feel the test would be more solid if it filled the payload with data and checked content on reception

Reply:
Good point. I modified the test to check the payload's contents.